firmware/application/apps/ui_fileman.cpp

```diff
@@ -238,79 +306,85 @@ FileLoadView::FileLoadView(
 	
 	refresh_list();
 	
-	on_select_entry = [&nav, this]() {
-		if (entry_list[menu_view.highlighted_index()].is_directory) {
-			load_directory_contents(get_selected_path());
+	on_select_entry = [this]() {
+		if (get_selected_entry().is_directory) {
+			load_directory_contents(get_selected_full_path());
 			refresh_list();
 		} else {
 			nav_.pop();
 			if (on_changed)
-				on_changed(current_path.string() + '/' + entry_list[menu_view.highlighted_index()].entry_path.string());
+				on_changed(get_selected_full_path());
 		}
 	};
 }
 
-void FileManagerView::on_rename(NavigationView& nav) {
-	text_prompt(nav, name_buffer, max_filename_length, [this](std::string& buffer) {
-		std::string destination_path = current_path.string();
-		if (destination_path.back() != '/')
-			destination_path += '/';
-		destination_path = destination_path + buffer;
-		rename_file(get_selected_path(), destination_path);
-		load_directory_contents(current_path);
-		refresh_list();
-	});
-}
+void FileManagerView::on_rename() {
+	// Don't allow rename of ".."
+	if (get_selected_entry().path == parent_dir_path)
+		return;
+
+	auto& entry = get_selected_entry();
+	auto name = entry.path.filename().string();
+	auto orig_ext = entry.path.extension();
+
+	uint32_t cursor_pos = (uint32_t)name.length();
+	if (auto pos = name.find_last_of("."); pos != name.npos)
+		cursor_pos = pos;
+
+	text_prompt(nav_, name, cursor_pos, max_filename_length,
```

Comment:
I'm getting garbage here in the textview control, but it seems like name.length() is correct because the cursor is where I would expect it to be.

Reply:
This was caused because the string was on the stack and was destroyed before the view was loaded.